MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -2041,27 +2042,32 @@ public virtual Task UpdateToRepositoryAsync(ItemUpdateType updateReason, Cancell
         /// <summary>
         /// Validates that images within the item are still on the filesystem.
         /// </summary>
-        /// <param name="directoryService">The directory service to use.</param>
         /// <returns><c>true</c> if the images validate, <c>false</c> if not.</returns>
-        public bool ValidateImages(IDirectoryService directoryService)
+        public bool ValidateImages()
         {
-            var allFiles = ImageInfos
-                .Where(i => i.IsLocalFile)
-                .Select(i => System.IO.Path.GetDirectoryName(i.Path))
-                .Distinct(StringComparer.OrdinalIgnoreCase)
-                .SelectMany(path => directoryService.GetFilePaths(path))
-                .ToList();
+            List<ItemImageInfo> deletedImages = null;
+            foreach (var imageInfo in ImageInfos)
+            {
+                if (!imageInfo.IsLocalFile)
+                {
+                    continue;
+                }
 
-            var deletedImages = ImageInfos
-                .Where(image => image.IsLocalFile && !allFiles.Contains(image.Path, StringComparison.OrdinalIgnoreCase))
-                .ToList();
+                if (File.Exists(imageInfo.Path))
```

Comment:
This might be worse for network storage (as DirectoryService pulled all paths from the containing folder) but idk